src/swbf_fnv_hashes.cpp

```diff
@@ -1,2834 +1,3935 @@
-   "WaitMin"_fnvp,
-   "WakeEffect"_fnvp,
-   "WakeWaterSplashEffect"_fnvp,
-   "walker"_fnvp,
-   "walkerdroid"_fnvp,
-   "WalkerLegPair"_fnvp,
-   "WalkerOrientRoll"_fnvp,
-   "WALKERSECTION"_fnvp,
-   "WalkerWidth"_fnvp,
-   "WarmUpSound"_fnvp,
-   "WarmUpSoundFadeInTime"_fnvp,
-   "WarmUpSoundFadeOutTime"_fnvp,
-   "WarmUpSoundStartPitch"_fnvp,
-   "WarmUpTime"_fnvp,
-   "water"_fnvp,
-   "water_foley"_fnvp,
-   "WaterDamageAmount"_fnvp,
-   "WaterDamageInterval"_fnvp,
-   "WaterEffect"_fnvp,
-   "WaterInfo"_fnvp,
-   "WaterRingColor"_fnvp,
-   "WaterSplashColor"_fnvp,
-   "WaterSplashEffect"_fnvp,
-   "WaterWadingSound"_fnvp,
-   "WaterWakeColor"_fnvp,
-   "WaverRate"_fnvp,
-   "WaverTurn"_fnvp,
-   "weapon"_fnvp,
-   "Weapon"_fnvp,
-   "WeaponAmmo1"_fnvp,
-   "WeaponAmmo2"_fnvp,
-   "WeaponAmmo3"_fnvp,
-   "WeaponAmmo4"_fnvp,
-   "WeaponAmmo"_fnvp,
-   "WeaponChange"_fnvp,
-   "WeaponChangeSound"_fnvp,
-   "WeaponChannel1"_fnvp,
-   "WeaponChannel2"_fnvp,
-   "WeaponChannel3"_fnvp,
-   "WeaponChannel4"_fnvp,
-   "WeaponChannel"_fnvp,
-   "WeaponName1"_fnvp,
-   "WeaponName2"_fnvp,
-   "WeaponName3"_fnvp,
-   "WeaponName4"_fnvp,
-   "WeaponName"_fnvp,
-   "WEAPONSECTION"_fnvp,
-   "WeaponSection"_fnvp,
-   "WeaponShareAmmo1"_fnvp,
-   "WeaponShareAmmo2"_fnvp,
-   "WeaponShareAmmo3"_fnvp,
-   "WeaponShareAmmo4"_fnvp,
-   "WeaponShareAmmo"_fnvp,
-   "WeaponShareHeat1"_fnvp,
-   "WeaponShareHeat2"_fnvp,
-   "WeaponShareHeat3"_fnvp,
-   "WeaponShareHeat4"_fnvp,
-   "WeaponShareHeat"_fnvp,
-   "WeaponUnableSelect"_fnvp,
-   "WheelOmegaToU"_fnvp,
-   "WheelOmegaToV"_fnvp,
-   "WheelSection"_fnvp,
-   "WheelTexture"_fnvp,
-   "WheelVelocToU"_fnvp,
-   "WheelVelocToV"_fnvp,
-   "Widescreen"_fnvp,
-   "Width"_fnvp,
-   "WiggleAmount"_fnvp,
-   "WiggleSpeed"_fnvp,
-   "win"_fnvp,
-   "winddirection"_fnvp,
-   "WindDownSound"_fnvp,
-   "WindSound"_fnvp,
-   "windspeed"_fnvp,
-   "WindUpSound"_fnvp,
-   "WingModel"_fnvp,
-   "word"_fnvp,
-   "word_reformat"_fnvp,
-   "WorldAligned"_fnvp,
-   "WorldBrightness"_fnvp,
-   "WorldContrast"_fnvp,
-   "WorldFogRange"_fnvp,
-   "WorldHeight"_fnvp,
-   "WorldRange"_fnvp,
-   "WorldSaturation"_fnvp,
-   "WrapInput"_fnvp,
-   "YawLimits"_fnvp,
-   "YawSpread"_fnvp,
-   "YawTurnFactor"_fnvp,
-   "YOffset"_fnvp,
-   "ZeroGravity"_fnvp,
-   "ZoomFirstPerson"_fnvp,
-   "ZoomMax"_fnvp,
-   "ZoomMin"_fnvp,
-   "ZoomRate"_fnvp,
-   "ZoomTurnDivisorMax"_fnvp,
-   "ZoomTurnDivisorMin"_fnvp,
-   "ZOrder"_fnvp};
-}
+std::mutex dictionary_mutex;
 
 std::string lookup_fnv_hash(const std::uint32_t hash)
 {
+   std::string retVal;
+
+   dictionary_mutex.lock();
    const auto result = swbf_hashes.find(hash);
 
-   if (result != std::cend(swbf_hashes)) return std::string{result->second};
+   if (result != std::cend(swbf_hashes)) {
+      // this can be interesting when building a dict
+      //synced_cout::print("looked_up:", result->second, "\n");
+      retVal = std::string{result->second};
+   }
+   dictionary_mutex.unlock();
```

Comment:
We should use [std::shared_lock](https://en.cppreference.com/w/cpp/thread/shared_mutex) rather than directly interacting with the mutex, this way we can be confident we never forgot to unlock it.
```suggestion
   {
      std::shared_lock lock{swbf_hashes_mutex};

      if (const auto result = swbf_hashes.find(hash); result != std::cend(swbf_hashes)) {
         return result->second;
      }
   }
```

Reply:
Please review the mutex handling here to ensure I understood your suggestion here